wrapanapi/msazure.py

```diff
@@ -359,8 +359,50 @@ def list_stack(self, resource_group=None, days_old=0):
     def list_flavor(self):
         raise NotImplementedError('list_flavor not implemented.')
 
-    def list_network(self):
-        raise NotImplementedError('list_network not implemented.')
+    def list_networks(self):
+        self.logger.info("Attempting to List Azure Virtual Private Networks of the provider region")
+        # Azure API returns all networks from all regions, and there is options to filter by region.
+        # In CFME only the networks of the provider regions are displayed.
+        all_networks = self.network_client.virtual_networks.list_all()
+        networks_in_region = []
+        location = self.region.replace(' ', '').lower()
+        for network in all_networks:
+            if network.location == location:
+                networks_in_region.append(network.name)
+        return networks_in_region
+
+    def list_subnets(self):
+        self.logger.info("Attempting to List Azure Subnets")
+        # There is no way to list all the subnets from a network filtered by location, and there
+        # is only one network in the resource_group defined in cfme_data.
+        all_networks = self.network_client.virtual_networks.list_all()
+        location = self.region.replace(' ', '').lower()
+        subnets = []
+        for network in all_networks:
+            if network.location == location:
```

Comment:
Up until this point here, this code block looks the same as list_networks() above. Could you simply do `all_networks = self.list_networks` and then iterate on that to get subnets?

Reply:
Yes it is similar, the problem is that then, I would need to use virtual_networks.get(), which take also as parameter a resource_group which is not always the same, and only 'Automation' is defined in the yaml.

Reply:
True, forgot you're only returning the name